### PR TITLE
Paint 1.0

### PR DESCRIPTION
**Pull request by deioo**
### What's New

* Simplified file structure _(core, tools, global, components)_
* Clearly define hierarchy based on component function and dependencies _(core functions are standalone, tools only use core, global only use tools and core and so on.)_
* Unified output where component settings getter function is used consistently, instead of combining functions and variables to return certain values (e.g use `radius(round)` instead of `$border-radius-rounded`, or `font(weight, bold)` instead of `$font-weight-bold`)
* Removed BEM naming conventions from base components to avoid confusion (e.g instead of modifiers like `%row--middle` we now have `%row-middle`)
* The only places where BEM structure is used are the more complex components like `button-group` or `notifications`, where we have to name specific elements like `button-group__item` or `notification__action`.
* Remove single-property 'atoms' and allow everyone to call the CSS property with a settings value as a param instead _(this avoids creating an unnecessary abstraction)_ E.g %border__bottom is now border-bottom: 1px solid color(border);
* New flex grid 
* Simplified buttons that include only `base`, `primary` and `secondary` instead of success, warning, danger, info and inverted styles.

### File structure

* Moved stylesheets into CORE, TOOLS, GLOBAL and COMPONENTS based on dependency
* Created a deprecated components graveyard, with code that is to be ported over to the apps
* Removed global variables, making all settings component-specific
* Removed external library dependencies _(like font-awesome)_

### Component structure

* Follow the same content structure for each and every stylesheet:
  * Settings map / function to retrieve settings
  * Private and public mixins
  * Private placeholders
  * Exports mixin that encapsulates dynamically generated placeholders to avoid duplicating the logic in case of multiple inclusion

### CODE REVIEW

**Comment by davegriffiths** [comment]
screen shot 2016-08-11 at 10 24 24

This is too big to review well, can we split it up?

**Comment by deioo** [comment]
I don't really see how, unless we stack different PRs on top of each other.
Can review individual commits to follow the changes better?

**Comment by davegriffiths** [comment]
Well we can split out the DSL changes, the glue code, the visual changes and the bug fixes?

This way we can add full descriptions to the PR's (especially the DSL changes).

**Comment by deioo** [comment]
The whole glue code is inside the app, but when the release is going out it will contain a Changelog _(same as all other important releases)_. I can start writing the changelog now if you think it would be useful for the review.
Also we can pair-review instead, since it would make things a lot easier to follow?

File: README.md [hunk 1]
```diff
@@ -72,87 +66,44 @@ To make any future changes easier, add all of the above in a `paint-loader.scss`
...
-- dropdown
-- step
-```
-
-**Deprecation warning**  
-Most of these components are going to be extracted into separate repos and refactored to make it easier to customise and more flexible to extend.
+Paint is structured into 4 main sections:
+
+- /CORE - contains a set of functions and mixins that are mandatory for the rest of the paint components to work properly. It includes:
+  - export
```
Comment (davegriffiths) [inline README.md]: Link to more details.

File: README.md [hunk 2]
```diff
@@ -72,87 +66,44 @@ To make any future changes easier, add all of the above in a `paint-loader.scss`
...
+  - map
+  - color
+  - unit
+- /TOOLS - a set of reusable styles that we want to share between the components and throughout the application
+  - color
```
Comment (davegriffiths) [inline README.md]: Can we have a symbol or similar to denote functions versus placeholders etc

File: README.md [hunk 3]
```diff
@@ -72,87 +66,44 @@ To make any future changes easier, add all of the above in a `paint-loader.scss`
...
-**Deprecation warning**  
-Most of these components are going to be extracted into separate repos and refactored to make it easier to customise and more flexible to extend.
+Paint is structured into 4 main sections:
+
+- /CORE - contains a set of functions and mixins that are mandatory for the rest of the paint components to work properly. It includes:
+  - export
+  - map
+  - color
+  - unit
+- /TOOLS - a set of reusable styles that we want to share between the components and throughout the application
```
Comment (davegriffiths) [inline README.md]: "a collection of mixins, placeholders and functions that we want to use across all components and the application"

File: styles/components/_avatar.scss [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    base: rem(36),
+    small: rem(24),
+    large: rem(48)
+  )
+);
+
+$avatar: () !default;
+$avatar: map-merge-settings($avatar-defaults, $avatar);
+
+@function avatar($settings...) {
```
Comment (davegriffiths) [inline styles/components/_avatar.scss]: usage

```css
@extend %avatar
@extend %avatar-small
```

File: README.md (hunk 3 above)
Comment (davegriffiths) [inline README.md]: All the functions return values which can be assigned as CSS property values.

**Comment by deioo** [comment]
@davegriffiths ok to release?

**Comment by davegriffiths** [comment]
Can you add the buttons to the description :)

**Comment by davegriffiths** [comment]
But then yes  :+1:

**Comment by deioo** [comment]
Dun :100:

**Pull request merged**